subsys/pm/device_runtime.c

```diff
@@ -67,7 +67,7 @@ static int runtime_suspend(const struct device *dev, bool async)
 		goto unlock;
 	}
 
-	if (async && !k_is_pre_kernel()) {
+	if ((async && !k_is_pre_kernel()) || k_is_in_isr()) {
```

Comment:
I think we should not allow `pm_device_runtime_put` in ISR context, but force usage of `pm_device_runtime_put_async` and document it with `@funcprops \isr_ok`.

Reply:
What about instead adding a note for `pm_device_runtime_put` like:
```
@note Synchronous put operation is not supported when in an ISR. In this 
case, the function will be asynchronous (equivalent to                   
pm_device_runtime_put_async()).                                          
```
marking `pm_device_runtime_put_async` with `\isr_ok`?

Reply:
IMO it's better to explicitely ask users to use async call in ISR context, so it is clear that is a deferred operation. Actually, ISRs were one usecase for _async call, even though it was broken.

Reply:
added an `__ASSERT` for that